benchmarks/graphics_pipeline/main.cpp

```diff
@@ -549,37 +646,44 @@ void ProjApp::Setup()
                 // Model matrix to be applied to the sphere mesh
                 float4x4 modelMatrix = glm::translate(float3(x * grid.step, y * grid.step, z * grid.step));
 
-                // Copy a sphere mesh to create a giant vertex buffer
-                // Iterate through the meshes vertx data and add it to the geometry
+                // Overwrite position data
                 for (uint32_t vertexIndex = 0; vertexIndex < sphereVertexCount; ++vertexIndex) {
                     TriMeshVertexData vertexData = {};
                     mesh.GetVertexData(vertexIndex, &vertexData);
                     vertexData.position = modelMatrix * float4(vertexData.position, 1);
 
                     TriMeshVertexDataCompressed vertexDataCompressed;
                     vertexDataCompressed.position = half3(glm::packHalf1x16(vertexData.position.x), glm::packHalf1x16(vertexData.position.y), glm::packHalf1x16(vertexData.position.z));
-                    vertexDataCompressed.texCoord = half2(glm::packHalf1x16(vertexData.texCoord.x), glm::packHalf1x16(vertexData.texCoord.y));
-                    vertexDataCompressed.normal   = i8vec4(MapFloatToInt8(vertexData.normal.x), MapFloatToInt8(vertexData.normal.y), MapFloatToInt8(vertexData.normal.z), MapFloatToInt8(1.0f));
-                    vertexDataCompressed.tangent  = i8vec4(MapFloatToInt8(vertexData.tangent.x), MapFloatToInt8(vertexData.tangent.y), MapFloatToInt8(vertexData.tangent.z), MapFloatToInt8(vertexData.tangent.a));
-                    lowPrecisionInterleaved.AppendVertexData(vertexDataCompressed);
-                    lowPrecisionPositionPlanar.AppendVertexData(vertexDataCompressed);
-
-                    highPrecisionInterleaved.AppendVertexData(vertexData);
-                    highPrecisionPositionPlanar.AppendVertexData(vertexData);
+
+                    OverwritePositionData(lowInterleavedPositionBuffer, vertexDataCompressed, i * sphereVertexCount + vertexIndex);
```

Comment:
It's a little hard to understand the overall picture of what's happening and what's important, perhaps breaking into separate functions will help? or adding comments at the top or in place will help?

Reply:
Added more comments to this section along with diagrams, let me know if it's still unclear!